lib/debug/session.rb

```diff
@@ -428,8 +428,24 @@ def process_command line
 
       # * `bt` or `backtrace`
       #   * Show backtrace (frame) information.
+      # * `bt <num>`
+      #   * Only shows first `<num>` frames.
+      # * `bt /location_regexp/`
+      #   * Only shows frames with location info that matches `/location_regexp/`.
+      # * `bt <num> /location_regexp/`
+      #   * Only shows first `<num>` frames with location info that matches `/location_regexp/`.
       when 'bt', 'backtrace'
-        @tc << [:show, :backtrace]
+        case arg
+        when /\A(\d+)\z/
+          @tc << [:show, :backtrace, arg.to_i]
+        when /\A\/.*\/\z/
+          @tc << [:show, :backtrace, eval(arg)]
```

Comment:
do not use `eval`, but use `Regexp.compile`.

Reply:
The message format should be `[:show, :backtrace, num, filter_regexp]`.
Do not mix the types.

Reply:
and simply `nil` represents "not specified".